TYPO3.Flow/Classes/TYPO3/Flow/Composer/InstallerScripts.php

```diff
@@ -62,18 +62,40 @@ static public function postPackageUpdateAndInstall(PackageEvent $event) {
 			}
 
 			$installPath = $event->getComposer()->getInstallationManager()->getInstallPath($package);
-			if (isset($packageExtraConfig['typo3/flow']['manage-resources']) && $packageExtraConfig['typo3/flow']['manage-resources'] === TRUE) {
-				if (is_dir($installPath . 'Resources/Private/Installer/Distribution/Essentials')) {
-					Files::copyDirectoryRecursively($installPath . 'Resources/Private/Installer/Distribution/Essentials', getcwd() . '/', FALSE, TRUE);
-				}
-				if (is_dir($installPath . 'Resources/Private/Installer/Distribution/Defaults')) {
-					Files::copyDirectoryRecursively($installPath . 'Resources/Private/Installer/Distribution/Defaults', getcwd() . '/', TRUE, TRUE);
+
+			$evaluatedInstallerResources = FALSE;
+			if (isset($packageExtraConfig['neos']['installer-resource-folders'])) {
+				foreach ($packageExtraConfig['neos']['installer-resource-folders'] as $installerResourceDirectory) {
+					static::copyDistributionFiles($installPath . $installerResourceDirectory);
 				}
+				$evaluatedInstallerResources = TRUE;
+			}
+
+			if (!$evaluatedInstallerResources && isset($packageExtraConfig['typo3/flow']['manage-resources']) && $packageExtraConfig['typo3/flow']['manage-resources'] === TRUE) {
```

Comment:
would be a else if or a nested if in the else, seemed to be more complicate to read.